### ISSUE TITLE
ApacheKiller DoS - Range Header Too Man Fields

### ISSUE THREAD

**Issue by fzipi** [crstest01/owasp-modsecurity-crs#229]
_Issue originally created by user mailtoyogeshpatel on date 2015-05-15 09:46:15.
            Link to original issue: https://github.com/SpiderLabs/owasp-modsecurity-crs/issues/229._

In our scenario we could not determine which document has a problem and blocked by mod security.
I can only say that we are opening pdf file from browser.I am not sure which type of documents has such bytes range which violets the rule.
How to identify such range and what i need to change in rule to allow more range.

**Comment by fzipi** [comment]
_User mailtoyogeshpatel commented on date 2015-05-19 13:33:13:_

 In our application we are getting "ApacheKiller DoS - Range Header Too many fields" in modsecurity log.

```
  As we are unable to recognisze the same, can you please tell us what could be the root cause  and what action should be taken to resolve this issue.
```

 See the attached image.

![dos_attack](https://cloud.githubusercontent.com/assets/11291299/7704055/a8e7ca48-fe59-11e4-8a28-709ee9ebdb64.jpg)

**Comment by fzipi** [comment]
_User csanders-git commented on date 2016-07-14 05:06:18:_

It'd be nice to have an example document. We are working to improve these rules and it is good to know browsers may split things up like this, although it is pretty unclear why in this case there are so many range fields.

**Comment by fzipi** [comment]
_User marcstern commented on date 2016-07-14 06:47:16:_

This is due to a PDF plugin performing (pseudo) streaming - displaying pages before the whole PDF is downloaded.

**Comment by fzipi** [comment]
_User dune73 commented on date 2016-11-19 09:08:35:_

The Core Rule Set 3 has moved this rule out of the default installation. If you raise the Paranoia Level setting to 2, the rule is back, but with a limit of 34 (!) ranges for the range header (that is the maximum we have seen in the wild by legitimate clients). In Paranoia Level 4, the limit of 5 ranges is enforced for PDFs again.

Closing this.
